src/capi/azext_capi/custom.py

```diff
@@ -384,9 +388,38 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
         ephemeral_disks=False,
         windows=False,
         pivot=False,
-        output_path=None,
+        template=None,
         yes=False):
 
+    if template:
+        mutual_exclusive_args = [
+            {
+                "name": "external_cloud_provider",
+                "value": external_cloud_provider
+            },
+            {
+                "name": "vnet_name",
+                "value": vnet_name
+            },
+            {
+                "name": "machinepool",
+                "value": machinepool
+            },
+            {
+                "name": "ephemeral_disks",
+                "value": ephemeral_disks
+            },
+            {
+                "name": "windows",
+                "value": windows
+            }
+        ]
+        defined_args = [v["name"] for v in mutual_exclusive_args if v["value"]]
+        if len(defined_args):
```

Comment:
```suggestion
        if defined_args:
```

Lists in Python evaluate as `True` if they're not empty, so you can skip the `len()` here.

Reply:
Thank you for this catch. My old JS instincts showing up lol